### PR TITLE
support refresh_token in resource_client_auth0

### LINKED ISSUE #19: refresh_token field for client
### Description

I noticed the provider does not support (AFAIK) the refresh token rotation for clients
In https://auth0.com/docs/api/management/v2#!/Clients/get_clients, we see there is an added refresh_token field, with this kind of format:
    "refresh_token": {
      "rotation_type": "non-rotating",
      "expiration_type": "non-expiring",
      "leeway": 0,
      "token_lifetime": 2592000
    }
Expect a PR, I guess, if i manage to write some go. Wish me luck ^^

### New or Affected Resource(s)

* auth0_client

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by rienafairefr**
### Proposed Changes

This adds support for te `refresh_token` field on clients

Fixes #19 

#### Acceptance Test Output

```
$ make testacc TESTS=TestAccClient.*
==> Checking that code complies with gofmt requirements...
?   	github.com/terraform-providers/terraform-provider-auth0	[no test files]
=== RUN   TestAccClientGrant
--- PASS: TestAccClientGrant (22.34s)
=== RUN   TestAccClient
--- PASS: TestAccClient (7.24s)
=== RUN   TestAccClientZeroValueCheck
--- PASS: TestAccClientZeroValueCheck (8.34s)
=== RUN   TestAccClientRotateSecret
--- PASS: TestAccClientRotateSecret (1.79s)
=== RUN   TestAccClientInitiateLoginUri
--- PASS: TestAccClientInitiateLoginUri (0.02s)
=== RUN   TestAccClientJwtScopes
--- PASS: TestAccClientJwtScopes (7.82s)
PASS
coverage: 26.0% of statements
ok  	github.com/terraform-providers/terraform-provider-auth0/auth0	47.560s	coverage: 26.0% of statements
?   	github.com/terraform-providers/terraform-provider-auth0/auth0/internal/debug	[no test files]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok  	github.com/terraform-providers/terraform-provider-auth0/auth0/internal/random	0.009s	coverage: 0.0% of statements [no tests to run]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok  	github.com/terraform-providers/terraform-provider-auth0/auth0/internal/validation	0.001s	coverage: 0.0% of statements [no tests to run]
?   	github.com/terraform-providers/terraform-provider-auth0/version	[no test files]

```

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

File: auth0/resource_auth0_client_test.go [hunk 1]
```diff
@@ -88,7 +93,7 @@ resource "auth0_client" "my_client" {
   is_first_party = true
   is_token_endpoint_ip_header_trusted = true
   token_endpoint_auth_method = "client_secret_post"
-  oidc_conformant = false
+  oidc_conformant = true
```
Comment (alexkappa) [inline auth0/resource_auth0_client_test.go:96]: Can we revert this? I think CI might break with this set to false due to the tenant configuration...

**Review by alexkappa — COMMENTED** [review]
Excellent work here @rienafairefr thank you!

File: auth0/resource_auth0_client_test.go (hunk 1 above)
Comment (rienafairefr) [reply auth0/resource_auth0_client_test.go:96]: can't really revert, i think. When it's set to false, I get this during the testacc run:
"Error: 400 Bad Request: Application must be OIDC Conformant when Refresh Token rotation is enabled"

**Comment by ninja-** [comment]
why was this PR closed? can we merge this fix?

**Comment by alexkappa** [comment]
It has been moved to https://github.com/alexkappa/terraform-provider-auth0/pull/255 as its the point we track the project issues/PRs.

**Pull request closed without merging**